byzcoin/service_test.go

```diff
@@ -513,6 +514,76 @@ func TestService_AddTransactionVersion(t *testing.T) {
 	require.Contains(t, reply.Error, "wrong byzcoin version")
 }
 
+func TestService_AutomaticVersionUpgrade(t *testing.T) {
+	// Creates a chain starting with version 0.
+	s := newSerWithVersion(t, 1, testInterval, 4, disableViewChange, 0)
+	defer s.local.CloseAll()
+
+	closing := make(chan bool)
+	wg := sync.WaitGroup{}
+	go func(closeChan chan bool) {
+		wg.Add(1)
+		defer wg.Done()
+
+		c := uint64(1)
+		shutdown := false
+		wait := 0
+		for !shutdown {
+			select {
+			case <-closing:
+				shutdown = true
```

Comment:
Just out of curiosity: why doesn't this just `return`?

Reply:
because I wanted the last transaction to take the `wait = 10` so everything closes correctly.